strawberry/types/type_resolver.py

```diff
@@ -138,23 +122,25 @@ class if one is not set by either using an explicit strawberry.field(name=...) o
             # Note: We do this here rather in the `Strawberry.type` setter
             # function because at that point we don't have a link to the object
             # type that the field as attached to.
-            if isinstance(field.type_annotation, StrawberryAnnotation):
-                type_annotation = field.type_annotation
-                type_annotation.annotation = _resolve_specialized_type_var(
-                    cls,
-                    cast(type, type_annotation.annotation),
-                )
-                if type_annotation.namespace is None:
-                    type_annotation.set_namespace_from_field(field)
+            if (
+                isinstance(field.type_annotation, StrawberryAnnotation)
+                and field.type_annotation.namespace is None
+            ):
+                field.type_annotation.set_namespace_from_field(field)
+
+            # If the field origin is not cls, copy it and set cls on it.
+            # This is important for us to be able to resolve specialized
+            # generic when generating the schema.
+            if field.origin != cls:
+                field = copy.copy(field)  # noqa: PLW2901
+                field.origin = cls
 
         # Create a StrawberryField for fields that didn't use strawberry.field
         else:
             # Only ignore Private fields that weren't defined using StrawberryFields
             if is_private(field.type):
                 continue
 
-            field_type = _resolve_specialized_type_var(cls, field.type)
-
             origin = origins.get(field.name, cls)
```

Comment:
is this not used anymore?

Reply:
ah sorry, I meant the `origin` variable 😊 below you're using `cls` instead of that (also why? :D)